lib/frr_zmq.c

```diff
@@ -330,12 +330,16 @@ void frrzmq_check_events(struct frrzmq_cb **cbp, struct cb_core *core,
 	len = sizeof(events);
 	if (zmq_getsockopt(cb->zmqsock, ZMQ_EVENTS, &events, &len))
 		return;
-	if (events & event && core->thread && !core->cancelled) {
+	if ((events & event) && core->thread && !core->cancelled) {
 		struct thread_master *tm = core->thread->master;
+
 		thread_cancel(&core->thread);
 
-		thread_add_event(tm, (event == ZMQ_POLLIN ? frrzmq_read_msg
-							  : frrzmq_write_msg),
-				 cbp, cb->fd, &core->thread);
+		if (event == ZMQ_POLLIN)
```

Comment:
White space change?

Reply:
well... there's macro expansion in there, and it doesn't like the embedded conditional, so I considered that part of "trouble with macros" ?